demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,386 @@
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m_td.empty() && FLAGS_m_tr.empty()) {
+        throw std::logic_error("Neither parameter -m_td nor -m_tr is not set");
+    }
+
+    if (!FLAGS_m_tr.empty() && FLAGS_dt.empty()) {
+        throw std::logic_error("Parameter -dt is not set");
+    }
+    return true;
+}
+
+void setLabel(cv::Mat& im, const std::string& label, const cv::Point& p);
+int clip(int x, int maxVal) { return std::min(std::max(x, 0), maxVal); }
+
+int main(int argc, char *argv[]) {
+    try {
+        /** This demo covers certain topologies only **/
+        // Parsing and validating input arguments
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+        const auto inputPath     = FLAGS_i;
+        const bool noShow        = FLAGS_no_show;
+        const bool loop          = FLAGS_loop;
+        const bool gapiStreaming = !FLAGS_gapi_regular_mode;
+        const bool rawOutput     = FLAGS_r;
+        const auto outputPath        = FLAGS_o;
+        const uint outputFramesLimit = FLAGS_limit;
+
+        const auto tdModelPath = FLAGS_m_td;
+        const bool tdRequired  = !tdModelPath.empty();
+        const auto tdDevice    = FLAGS_d_td;
+
+        const auto trModelPath = FLAGS_m_tr;
+        const bool trRequired  = !trModelPath.empty();
+        const auto trDevice    = FLAGS_d_tr;
+        const auto trOutputBlobName = FLAGS_tr_o_blb_nm;
+        const bool trComposite = std::string::npos != trModelPath.find("encoder");
+        const std::array<std::string,2> encoderOutputNames { FLAGS_out_enc_hidden_name,
+                                                             FLAGS_features_name        };
+        const std::array<std::string,3> decoderInputNames { FLAGS_in_dec_symbol_name,
+                                                            FLAGS_in_dec_hidden_name,
+                                                            FLAGS_features_name         };
+        const std::array<std::string,2> decoderOutputNames { FLAGS_out_dec_hidden_name,
+                                                             FLAGS_out_dec_symbol_name  };
+        const auto decoderType       = FLAGS_dt;
+        const auto decoderBandwidth  = FLAGS_b;
+        const auto decoderStartIndex = FLAGS_start_index;
+        const size_t tdNewInputWidth  = FLAGS_w_td;
+        const size_t tdNewInputHeight = FLAGS_h_td;
+        const size_t tdMaxRectsNum    = FLAGS_max_rect_num;
+        const bool centralCrop        = FLAGS_cc;
+        const float segmConfThreshold = static_cast<float>(FLAGS_cls_pixel_thr);
+        const float linkConfThreshold = static_cast<float>(FLAGS_link_pixel_thr);
+        const double trMinConfidence  = FLAGS_thr;
+        const bool trPadSymbolFirst   = FLAGS_tr_pt_first;
```

Comment:
You can just read `FLAGS_*` themselves. No need to copy

Reply:
OK, removed unnecessary redefinitions